askama_shared/src/generator.rs

```diff
@@ -1367,6 +1421,60 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
         Ok(DisplayWrap::Unwrapped)
     }
 
+    fn visit_localize(
+        &mut self,
+        buf: &mut Buffer,
+        message: &str,
+        attribute: Option<&str>,
+        args: &[(&str, Expr)],
+    ) -> Result<DisplayWrap, CompileError> {
+        /* TODO
+        if !cfg!(feature = "with-i18n") {
+            panic!(
+                "The askama feature 'with-i18n' must be activated to enable calling `localize`."
+            );
+        }
+        */
+
+        // TODO
+        let localizer = self.input.localizer.as_ref().expect(
+            "A template struct must have a member with the `#[localizer]` \
+             attribute that implements `askama::Localize` to enable calling the localize() filter",
+        );
+
+        let mut message = message.to_string();
+        if let Some(attribute) = attribute {
+            message.push_str(".");
+            message.push_str(attribute);
+        }
+
+        assert!(
+            message.chars().find(|c| *c == '"').is_none(),
+            "message ids with quotes in them break the generator, please remove"
+        );
+
+        self.localized_messages.insert(message.clone());
+
+        buf.write(&format!(
+            "self.{}.translate(\"{}\", &std::iter::FromIterator::from_iter(vec![",
```

Comment:
Here we need to be able to call the translate method. Currently it uses the Localizer object that has to be added to all Template structs.